### PR TITLE
refactor patch release management

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

## Context

In preparation for upgrade tests we need logic to find the latest major, minor and patch version of the predecessor of the current release. This refactoring prepares for the next steps required. I need to check if I am happy with that still but so far this was what I could come up with.

### CODE REVIEW

**Review by calvix — APPROVED** [review]
looks sane to me :sweat_smile:

**Review by njuettner — APPROVED** [review]

File: pkg/action/cl001/ac001/crs.go [hunk 1]
```diff
@@ -17,31 +14,30 @@ import (
...
-	fmt.Println(r.Version())
 	{
 		c := v1alpha2.ClusterCRsConfig{
 			Credential:        key.Credential,
 			Domain:            key.DomainFromHost(host),
 			Description:       explainerCommand,
 			Owner:             key.Organization,
 			Region:            key.RegionFromHost(host),
-			ReleaseComponents: r.Components(),
-			ReleaseVersion:    strings.Replace(r.Version(), "v", "", -1),
```
Comment (xh3b4sd) [inline pkg/action/cl001/ac001/crs.go:44]: @njuettner I changed the approach. I would favour using the interface in a clean way if there is only this particular use case.

File: pkg/action/cl001/ac005/crs.go [hunk 2]
```diff
@@ -53,8 +53,8 @@ func newCRs(ctx context.Context, releases []v1alpha1.Release, id string, host st
 			OnDemandBaseCapacity:                0,
 			OnDemandPercentageAboveBaseCapacity: 0,
 			Owner:                               "giantswarm",
-			ReleaseComponents:                   r.Components(),
-			ReleaseVersion:                      r.Version(),
+			ReleaseComponents:                   p.Components(),
+			ReleaseVersion:                      p.Version(),
```
Comment (xh3b4sd) [inline pkg/action/cl001/ac005/crs.go:57]: This here was maybe broken as well, but should be fixed now.

File: pkg/release/patch.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+		for _, c := range p.release().Spec.Components {
+			releaseComponents[c.Name] = c.Version
+		}
+	}
+
+	return releaseComponents
+}
+
+func (p *Patch) Version() string {
+	return strings.Replace(p.release().GetName(), "v", "", 1)
```
Comment (xh3b4sd) [inline pkg/release/patch.go:64]: This is how it is uniformly fixed now.

**Pull request merged**